### DISCUSSION TITLE
'Within' keyword with '-Reason' parameter is not working

### DISCUSSION THREAD

**Discussion by ankur90Git** [microsoft/PSRule#1116] [Q&A] [answered]
I am using 'Within' keyword inside a rule along with parameter '-Reason' to pass the custom message on failure. But in the output the reason is blank for the failed rule. How can I get the custom failure message in the results?

```powershell
# Synopsis: key FUNCTIONS_EXTENSION_VERSION must have value of pattern '~1'
Rule 'FUNCTIONS_EXTENSION_VERSION1'  {
    
    AllOf { 
        Exists 'ApplicationSettings.FUNCTIONS_EXTENSION_VERSION' -CaseSensitive -Reason 'field not present'
        Match 'ApplicationSettings.FUNCTIONS_EXTENSION_VERSION' '7' -Reason 'value not matching as expected'
    }
}
```

```powershell
Invoke-PSRule -Path $rulePath -InputPath $inputPath -OutputFormat wide
```

![image](https://user-images.githubusercontent.com/84568209/174335292-80c06c5e-22da-4c52-b90c-569fcaedbcac.png)

**Comment by BernieWhite** [comment, marked as answer]
@ankur90Git Yes this is an issue with `Invoke-PSRule` thanks for reporting the issue. I've logged it as a bug #1117.

The reason is there but there is an issue with wide view which is preventing the display of the property.

A temporary workaround would be:

```powershell
Invoke-PSRule -Path $rulePath -InputPath $inputPath -OutputFormat wide | Format-Table RuleName,Outcome,@{ Name = 'Reason'; Expression = { $_.Reason } },Recommendation
```
